### ISSUE TITLE
Development installation can't progress on Ubuntu 20.04 Focal Fossa

### ISSUE THREAD

**Issue by radiodevx** [AzuraCast/AzuraCast#4553]
### Installation Method

Docker Installation

### AzuraCast Release Channel

Rolling Release Channel

### Current AzuraCast Version

Rolling release #d37216d

### What happened?

On a clean Ubuntu 20.04 Focal Fossa I tried to make development work but When I try to do it I always get this errors.

First;

I am doing as [this](https://docs.azuracast.com/en/developers/getting-started) page says. after I do `bash docker.sh install-dev
` I get a lot of autoload not found and bin/console not found errors as you can see in log output section.
After that when I try to get any view of the build I only get 502. Doing `docker-compose down/up -d` changes nothing. I tried this setup on multiple different ubuntu OS/windows/Macos and specs and I always get the same errors. I suspect that development documentation is not clear enough or the script that I am running is not fully working. Any solutions to this issue? How do you get it working for any kind of OS?
<img width="1279" alt="PHP Error Log" src="https://user-images.githubusercontent.com/74149660/132510506-ff1b3c97-519f-448f-816a-4c0d438eba13.png">
<img width="1279" alt="Frontent Build Error Log" src="https://user-images.githubusercontent.com/74149660/132510524-cdc7a7c3-7de4-4ef6-8dd4-66c921f19ba9.png">

### Relevant log output

```shell
#Service Startup Errors

Creating azuracast_redis    ... done
Creating azuracast_mariadb  ... done
Creating azuracast_web_run  ... done
                               Starting up AzuraCast services...
Services started up and ready!
AzuraCast Setup
(Environment: development)
Installing dependencies from lock file (including require-dev)
Verifying lock file contents can be installed on current platform.
Package operations: 204 installs, 0 updates, 0 removals


  [RuntimeException]
  /var/azuracast/www/vendor/composer does not exist and could not be created.


install [--prefer-source] [--prefer-dist] [--prefer-install PREFER-INSTALL] [--dry-run] [--dev] [--no-suggest] [--no-dev] [--no-autoloader] [--no-scripts] [--no-progress] [--no-install] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--] [<packages>]...

PHP Warning:  require(/var/azuracast/www/vendor/autoload.php): Failed to open stream: No such file or directory in /var/azuracast/www/bin/console on line 9
PHP Stack trace:
PHP   1. {main}() /var/azuracast/www/bin/console:0

Warning: require(/var/azuracast/www/vendor/autoload.php): Failed to open stream: No such file or directory in /var/azuracast/www/bin/console on line 9

Call Stack:
    0.0003     392552   1. {main}() /var/azuracast/www/bin/console:0

PHP Fatal error:  Uncaught Error: Failed opening required '/var/azuracast/www/vendor/autoload.php' (include_path='.:/usr/share/php') in /var/azuracast/www/bin/console:9
Stack trace:
#0 {main}
  thrown in /var/azuracast/www/bin/console on line 9

Fatal error: Uncaught Error: Failed opening required '/var/azuracast/www/vendor/autoload.php' (include_path='.:/usr/share/php') in /var/azuracast/www/bin/console on line 9

Error: Failed opening required '/var/azuracast/www/vendor/autoload.php' (include_path='.:/usr/share/php') in /var/azuracast/www/bin/console on line 9

Call Stack:
    0.0003     392552   1. {main}() /var/azuracast/www/bin/console:0

ERROR: 255
Building frontend
Sending build context to Doc

#Frontent build Error

   scss/style.scss 3:11                          root stylesheet

WARNING: 166 repetitive deprecation warnings omitted.

[11:46:17] 'scripts:jquery' errored after 21 s
[11:46:17] Error: EACCES: permission denied, mkdir '/data/web/static/dist/lib'
[11:46:17] 'bundle_deps' errored after 22 s
[11:46:17] 'default' errored after 22 s
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! azuracast@ build: `gulp`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the azuracast@ build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2021-09-08T11_46_17_341Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! azuracast@ dev-build: `npm ci && npm run build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the azuracast@ dev-build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/node/.npm/_logs/2021-09-08T11_46_17_478Z-debug.log
ERROR: 1
```

**Comment by MrABKhan** [comment]
I'm experiencing the same issue!

I tried getting the autoload.php from a running container in production but still couldn't work.

@SlvrEagle23 Currently I cannot setup any sort of development environment to work on Azuracast due to this issue.

**Comment by SlvrEagle23** [comment]
@radiodevx @MrABKhan It appears you're all experiencing some sort of permissions issue that I'm unable to reproduce in my own development environment.

Diagnosing this process is more difficult as my typical development environment is actually on Windows using the Docker Desktop utility, not on Ubuntu itself, though when setting up the latest `install-dev` script I was able to test that things worked fine on my installation of Pop!OS, which is based on Ubuntu 21.04.

Both of the errors in question, though, are related to permissions of subdirectories that should be created as volumes inside Docker.

I'll look into it as I have a chance to do so, but if anyone who is experiencing the issue is able to fix it, it will likely be done faster than I can get to it; definitely make sure both the `vendor` and `frontend/node_modules` folders have proper permissions.

**Comment by radiodevx** [comment]
Okay solution is changing the permissions. But when I tried to build after changing vendor and frontend it still gave error from web/static folder. 

As a little bash script I wrote this:

_Inside AzuraCast folder_
```bash
chmod 777 frontend/ web/ vendor/ web/static/ web/static/api/ web/static/dist/ web/static/icons/ web/static/icons/development/ web/static/icons/production/ web/static/img/
```
It worked with no problems on a fresh development installation.

Since this is not happening on all computers I would suggest adding this option as a warning to development documentation of this project.
